src/Underworld.ts

```diff
@@ -2125,6 +2125,22 @@ export default class Underworld {
     );
     return { x, y };
   }
+  getPlayerFactions() {
+    // Returns all players' current faction
+    return this.players.filter(p => p.clientConnected).map(p => p.unit.faction);
```

Comment:
I'd recommend deduplicating this array:
https://stackoverflow.com/a/9229821/4418836

so that it'd only ever be ['ally','3rd faction', 'etc'] rather than ['ally', 'ally', 'ally', '3rd faction', 'etc', 'etc']